### PR TITLE
Feat/kit com event

### PR DESCRIPTION
**Pull request by nkokla**

### CODE REVIEW

File: src/components/MarkdownWrapper/MarkdownWrapper.style.tsx [hunk 1]
```diff
@@ -51,9 +52,18 @@ export const MarkdownWrapperStyled = styled.div.attrs<MarkdownWrapperStyledProps
   }
 `;
 
-export const ButtonLink = styled.a.attrs((props) => ({
+type ExtraEventType = { trackEvent?: string };
+export const ButtonLink = styled.a.attrs((props: Record<string, any>) => ({
```
Comment (desoindx) [inline src/components/MarkdownWrapper/MarkdownWrapper.style.tsx:56]: Can't you list the available props ?

File: src/components/MarkdownWrapper/MarkdownWrapper.style.tsx [hunk 2]
```diff
@@ -51,9 +52,18 @@ export const MarkdownWrapperStyled = styled.div.attrs<MarkdownWrapperStyledProps
...
   ...props,
   className: `fr-btn ${props.className || ''}`,
+  ...((props as ExtraEventType)?.trackEvent && {
```
Comment (desoindx) [inline src/components/MarkdownWrapper/MarkdownWrapper.style.tsx:59]: weird to have a deconstructor here

File: src/components/MarkdownWrapper/MarkdownWrapper.style.tsx [hunk 3]
```diff
@@ -51,9 +52,18 @@ export const MarkdownWrapperStyled = styled.div.attrs<MarkdownWrapperStyledProps
...
+    onClick: () => {
+      matomoEvent(markupData.markdownEvent.matomoEvent, [
+        (props as any).trackEvent ||
```
Comment (desoindx) [inline src/components/MarkdownWrapper/MarkdownWrapper.style.tsx:62]: If you're here you have a trackEvent so the || is useless

File: src/components/Markup/markup-data.json [hunk 4]
```diff
@@ -27,5 +27,8 @@
   "contactFormIneligible": {
     "facebookEvent": ["Contact"],
     "matomoEvent": ["Eligibilité", "Formulaire de contact inéligible - Envoi"]
+  },
+  "markdownEvent": {
+    "matomoEvent": ["Text event", "Markdown"]
```
Comment (desoindx) [inline src/components/Markup/markup-data.json:32]: Text event and Markdown ? Can't you use a more precise text ?

**Review by desoindx — CHANGES_REQUESTED** [review]

File: src/components/MarkdownWrapper/MarkdownWrapper.style.tsx (hunk 3 above)
Comment (nkokla) [reply src/components/MarkdownWrapper/MarkdownWrapper.style.tsx:62]: The trackEvent can also be an empty string.

Comment (desoindx) [reply src/components/MarkdownWrapper/MarkdownWrapper.style.tsx:62]: then (props as ExtraEventType)?.trackEvent return false

Comment (nkokla) [reply src/components/MarkdownWrapper/MarkdownWrapper.style.tsx:62]: Ho, sure ! My bad.

File: src/components/MarkdownWrapper/MarkdownWrapper.style.tsx (hunk 2 above)
Comment (nkokla) [reply src/components/MarkdownWrapper/MarkdownWrapper.style.tsx:59]: I can't find it but if you prefer we can use an intermediate variable (personally, I find it heavier).

File: src/components/Markup/markup-data.json (hunk 4 above)
Comment (nkokla) [reply src/components/Markup/markup-data.json:32]: Sure. Could you give me the value your prefer to see appear in matomo ?

File: src/components/MarkdownWrapper/MarkdownWrapper.style.tsx [hunk 5]
```diff
@@ -51,10 +52,36 @@ export const MarkdownWrapperStyled = styled.div.attrs<MarkdownWrapperStyledProps
...
+  children: React.ReactNode;
+  className?: string;
+  tagName?: TagName;
+  trackEvent?: string;
+};
+
+export const ButtonLink = styled.a.attrs(
+  (props: React.HTMLProps<HTMLLinkElement> & ExtraEventType) => {
+    const { className, tagName, trackEvent } = props;
+    const getMatomoEventKey = (tagName?: TagName) =>
```
Comment (desoindx) [inline src/components/MarkdownWrapper/MarkdownWrapper.style.tsx:66]: Why are you creating a function for that ?

**Review by desoindx — APPROVED** [review]

File: src/components/MarkdownWrapper/MarkdownWrapper.style.tsx (hunk 5 above)
Comment (nkokla) [reply src/components/MarkdownWrapper/MarkdownWrapper.style.tsx:66]: Just for visibility.

**Pull request merged**